lightningnetwork/lnd
# Sign/Verify messages and signatures for single addresses

## Change Description
This PR adds grpc functionality for signing a message with an address which is solely based on a public key owned by the internal lnd wallet. It also adds the functionality to verify a base64 encoded signature with a provided address. 
The `lncli` program was also enhanced by the commands `signmessage` and `verifymessage` to allow using the new introduced grpc commands. 
It fixes #7202.

## Steps to Test
Itests where added to run use:

```
make itest icase=sign_verify_message_with_addr temptest=1
```
In addition it can be tested via the `lncli commands` manually
```
Create a new address
- lncli newaddress p2wkh
Sign a random message e.g.
- lncli wallet addresses signmessage  --address "own address" --msg "own message" 
Verify the returned signature with
- lncli wallet addresses verifymessage --address "address" --sig "base64 encoded signature"  --msg "message to verify" 
```

## Pull Request Checklist
### Testing

### Code Style and Documentation

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.
